src/cmd/build.js

```diff
@@ -40,41 +40,34 @@ export type PackageCreatorParams = {
 export type PackageCreatorFn =
     (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
 
-function defaultPackageCreator(
+async function defaultPackageCreator(
   {manifestData, sourceDir, fileFilter, artifactsDir}: PackageCreatorParams
 ): Promise<ExtensionBuildResult> {
+  let id;
 
-  return new Promise(
-    (resolve) => {
-      if (manifestData) {
-        const id = getManifestId(manifestData);
-        log.debug(`Using manifest id=${id || '[not specified]'}`);
-        resolve(manifestData);
-      } else {
-        resolve(getValidatedManifest(sourceDir));
-      }
-    })
-    .then((manifestData) => {
-      return zipDir(
-        sourceDir, {
-          filter: (...args) => fileFilter.wantFile(...args),
-        })
-        .then((buffer) => {
-          let packageName = safeFileName(
-            `${manifestData.name}-${manifestData.version}.zip`);
-          let extensionPath = path.join(artifactsDir, packageName);
-          let stream = createWriteStream(extensionPath);
-          let promisedStream = streamToPromise(stream);
-
-          stream.write(buffer, () => stream.end());
-
-          return promisedStream
-            .then(() => {
-              log.info(`Your web extension is ready: ${extensionPath}`);
-              return {extensionPath};
-            });
-        });
-    });
+  if (manifestData) {
+    id = getManifestId(manifestData);
+    log.debug(`Using manifest id=${id || '[not specified]'}`);
+  } else {
+    manifestData = await getValidatedManifest(sourceDir);
+  }
+
+  let buffer =  await zipDir(sourceDir, {
```

Comment:
I think the rule in question is [`no-multi-spaces`](http://eslint.org/docs/rules/no-multi-spaces).